scripts/stac/imagery/collection.py

```diff
@@ -0,0 +1,107 @@
+from datetime import datetime
+from typing import Any, Dict, List, Optional
+
+import ulid
+
+PYSTAC_VERSION = "1.0.0"
+
+
+class ImageryCollection:
+    stac: Dict[str, Any]
+
+    def __init__(
+        self, title: Optional[str] = None, description: Optional[str] = None, stac: Optional[Dict[str, Any]] = None
+    ) -> None:
+        if stac:
+            self.stac = stac
+        elif title and description:
+            self.stac = {
+                "type": "Collection",
+                "stac_version": PYSTAC_VERSION,
+                "id": str(ulid.ULID()),
+                "title": title,
+                "description": description,
+                "license": "CC-BY-4.0",
+                "links": [{"rel": "self", "href": "./collection.json", "type": "application/json"}],
+            }
+        else:
+            raise Exception("incorrect initialising parameters must have 'stac' or 'title and description'")
+
+    def add_link(self, href: str, rel: str = "item", file_type: str = "application/json") -> None:
+        self.stac["links"].append({"rel": rel, "href": href, "type": file_type})
+
+    def update_spatial_extent(self, item_bbox: List[float]) -> None:
+        if "extent" not in self.stac:
+            self.update_extent(bbox=item_bbox)
+            return
+        if not self.stac["extent"]["spatial"]["bbox"]:
+            self.update_extent(bbox=item_bbox)
+            return
+
+        bbox = self.stac["extent"]["spatial"]["bbox"]
+        min_x = min(bbox[0], bbox[2])
+        max_x = max(bbox[0], bbox[2])
+        min_y = min(bbox[1], bbox[3])
+        max_y = max(bbox[1], bbox[3])
+
+        item_min_x = min(item_bbox[0], item_bbox[2])
+        item_max_x = max(item_bbox[0], item_bbox[2])
+        item_min_y = min(item_bbox[1], item_bbox[3])
+        item_max_y = max(item_bbox[1], item_bbox[3])
+
+        if item_min_x < min_x:
+            min_x = item_min_x
+        if item_min_y < min_y:
+            min_y = item_min_y
+        if item_max_x > max_x:
+            max_x = item_max_x
+        if item_max_y > max_y:
+            max_y = item_max_y
+
+        self.update_extent(bbox=[min_x, min_y, max_x, max_y])
+
+    def update_temporal_extent(self, item_start_datetime: str, item_end_datetime: str) -> None:
```

Comment:
You could pass the `item` instead of these two parameters and manage getting the information from it in here

Reply:
as above comment